scripts/ImportExportTest.sh

```diff
@@ -108,44 +124,41 @@ function JsonEvmAsm_ImportExportEquivalence
                     do
                         for obtained_contract in $(jq '.contracts | keys | .[]' obtained.json  2> /dev/null)
                         do
-                            jq --raw-output ".contracts.${obtained_contract}.\"${output}\"" obtained.json > "obtained.${output}"
+                            jq --raw-output ".contracts.${obtained_contract}.\"${output}\"" obtained.json > "obtained.${output}.json"
                             set +e
-                            diff_files "expected.${output}" "obtained.${output}"
-                            DIFF=$?
-                            set -euo pipefail
-                            if [[ ${DIFF} != 0 ]]
+                            if ! diff_files "expected.${output}.json" "obtained.${output}.json"
                             then
                                 _TESTED=
                                 FAILED=$((FAILED + 1))
                                 return 0
                             fi
+                            set -e
                         done
                     done
 
                     # direct export via --asm-json, if imported with --import-asm-json.
-                    if ! $SOLC --asm-json --import-asm-json expected.asm | tail -n+4 > obtained_direct_import_export.json 2> obtained_direct_import_export.error
+                    if ! $SOLC --asm-json --import-asm-json expected.asm.json --pretty-json --json-indent 4 | tail -n+4 > obtained_direct_import_export.json 2> obtained_direct_import_export.error
                     then
                         printError
-                        printError "$nth_input_file"
+                        printError "$sol_file"
                         cat obtained_direct_import_export.error >&2
                         FAILED=$((FAILED + 1))
                         return 0
                     else
-                        for obtained_contract in $(jq '.contracts | keys | .[]' obtained_direct_import_export.json 2> /dev/null)
-                        do
-                            jq --raw-output ".contracts.${obtained_contract}.\"asm\"" obtained_direct_import_export.json > obtained_direct_import_export.asm
-                            set +e
-                            diff_files "expected.asm" "obtained_direct_import_export.asm"
-                            DIFF=$?
-                            set -euo pipefail
-                            if [[ ${DIFF} != 0 ]]
-                            then
-                                _TESTED=
-                                FAILED=$((FAILED + 1))
-                                return 0
-                            fi
-                        done
+                        # reformat jsons - yeah, somehow its not generating the same format
+                        # todo: fix that
+                        jq . expected.asm.json > expected.asm.json.pretty
```

Comment:
If it's just about indentation try `jq --indent=4`. Or are you seeing other differences?

Reply:
I did not investigate this further yet. I only saw that there was a diff regarding the arrays.
e.g.
```json
{
  "bla": [
    "a",
    "b"
  ]
}
```
and 
```json
{
  "bla": 
  [
    "a",
    "b"
  ]
}
```
Thats why I just reformat the output with `jq` here. I think there is something wrong with `--pretty-json`, but I did not look into this yet.

Reply:
No, both are valid ways to format this. In JSON whitespace does not matter so each formatter can do it its own way. solc uses json-cpp, which is apparently does not match `jq`.

Passing it through `jq` to get consistent formatting is fine.

Reply:
Ok! I leave it with reformatting using `jq`.